cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -369,11 +369,15 @@ func StartControllers(s *options.CMServer, kubeconfig *restclient.Config, rootCl
 	// Find the list of namespaced resources via discovery that the namespace controller must manage
 	namespaceKubeClient := client("namespace-controller")
 	namespaceClientPool := dynamic.NewClientPool(restclient.AddUserAgent(kubeconfig, "namespace-controller"), restMapper, dynamic.LegacyAPIPathResolverFunc)
```

Comment:
Dont you need to change the RESTMapper as well?
dynamic client has a NewDiscoveryRESTMapper method: https://github.com/kubernetes/kubernetes/blob/master/pkg/client/typed/dynamic/dynamic_util.go#L37